riscv/src/compiler.rs

```diff
@@ -756,40 +766,48 @@ fn process_instruction(instr: &str, args: &[Argument]) -> Vec<String> {
             // logical shift right
             let (rd, rs, amount) = rri(args);
             assert!(amount <= 31);
-            vec![format!("{rd} <=X= shr({rs}, {amount});")]
+            only_if_no_write_to_zero(format!("{rd} <=X= shr({rs}, {amount});"), rd)
         }
         "srl" => {
             // logical shift right
             let (rd, r1, r2) = rrr(args);
-            vec![
-                format!("tmp1 <=X= and({r2}, 0x1f);"),
-                format!("{rd} <=X= shr({r1}, tmp1);"),
-            ]
+            if rd.is_zero() {
+                vec![]
+            } else {
+                vec![
+                    format!("tmp1 <=X= and({r2}, 0x1f);"),
+                    format!("{rd} <=X= shr({r1}, tmp1);"),
```

Comment:
why not use the new function here too?

Reply:
Because it only takes a single argument.